### PR TITLE
Use gomod instead of godep and Update vendor packages

### PR DESCRIPTION
**Pull request by hiyosi**
The main purpose of this PR is updating the vendor packages.
Before this commits, the `gophercloud` included a bug(https://github.com/gophercloud/gophercloud/pull/1464). We encountered deadlock when re-authentication is proceeded.

Also we change vendoring tool at the same time as updating vendor packages.

### CODE REVIEW

**Comment by ryysud** [comment]
Could you also update CONTRIBUTING.md which depends on this PR change?
https://github.com/zlabjp/spire-openstack-plugin/blob/gomod/CONTRIBUTING.md

File: Makefile [hunk 1]
```diff
@@ -9,9 +9,6 @@ ifeq (${uname},Darwin)
 	OS=darwin
 endif
```
Comment (ryysud) [inline Makefile]: By declaring export command here, I think that `GO111MODULE=on` becomes unnecessary for each task.
```suggestion
# Enable go modules
export GO111MODULE=on
```

**Comment by hiyosi** [comment]
@summerwind @ryysud  Please review this PR.

File: Makefile [hunk 2]
```diff
@@ -9,8 +9,7 @@ ifeq (${uname},Darwin)
...
 
-utils = github.com/goreleaser/goreleaser \
-		github.com/golang/dep/cmd/dep
+export GO111MODULE=on
```
Comment (summerwind) [inline Makefile]: I'd like to suggest using module mirror by adding `GOPROXY=https://proxy.golang.org`
See: https://proxy.golang.org/

Comment (hiyosi) [reply Makefile]: Thanks for suggetion!
Module mirror seems fine, so I'll adopt it!

2b21a3a

**Pull request merged**